lesson11/main.js

```diff
@@ -0,0 +1,69 @@
+const ul = document.getElementById("js-ul");
+
+const addLoading = () =>{
+  const loading = document.getElementById("js-loading");
+  const loadingImg = document.createElement('img');
+  loadingImg.src = 'loading.gif';
+  loadingImg.id = "loading-img";
+  loading.appendChild(loadingImg);
+};
+
+const removeLoading = () =>{
+  document.getElementById('loading-img').remove();
+};
+
+const renderLists = (features) =>{
+  const fragment = new DocumentFragment();
+
+  for (let feature of features ){
+      const li = document.createElement('li');
+      const a = document.createElement('a');
+      const img = document.createElement('img');
+      a.href = feature.to;
+      a.textContent = feature.text;
+      img.src = feature.img;
+      img.alt = feature.alt;
+
+      fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
+    };
+
+    ul.appendChild(fragment);
+};
+
+const displayError = (error) =>{
+  const li = document.createElement('li');
+  li.textContent = error;
+  ul.appendChild(li);
+};
+
+
+
+const checkResponseStatus = async () => {
+  const fetchedData = await fetch('https://mocki.io/v1/ee8a871e-2b46-4a91-b565-4d6f9216f300');
+  if(fetchedData.status === 200){
+    return fetchedData.json();
+  } else {
+    return Promise.reject(new Error('Failed to fetch data from server!'));
+  }
+};
+
+
+const displayListOrError= async () =>{
+ await checkResponseStatus()
+ .then(json => {
+     renderLists(json.data);
+    })
+ .catch(error => {
+    displayError(error);
+  })
+};
+
+const init = () =>{
+  addLoading();
+  setTimeout(() => {
+    displayListOrError();
+    removeLoading();
+  }, 3000);
```

Comment:
MUST
Set timeout is not required from this assignment. I will share the feedback from Moriken Sensei with you.
[My GitHub](https://github.com/443n0511/js-lesson/pull/1#discussion_r579798767)

Reply:
Thank you for your review Yoshimi-san,
I removed setTimeout 

However, I could not managed to display loading image before the lists were rendered. 
Is loading img supposed to be displayed before the lists are rendered in this lesson?

Please check my latest commit 96057c52090197748e4cabff8d120d3c726ddf63

Reply:
> However, I could not managed to display loading image before the lists were rendered.

It would work if we implemented "delete loading image after successful fetch" with try-catch-finally.